src/App.tsx

```diff
@@ -1,25 +1,64 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import Pagination from './components/Pagination/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
+const items = getNumbers(1, total).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleChangePerPages = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const calculatePageRange = (
+    currentPageArg: number,
+    perPageArg: number,
+    totalArg: number,
+  ) => {
+    const startElement = (currentPageArg - 1) * perPageArg + 1;
+    const endElement = Math.min(currentPageArg * perPageArg, totalArg);
+
+    return { startElement, endElement };
+  };
```

Comment:
The `calculatePageRange` function is well-implemented and encapsulates the pagination logic effectively. However, consider moving this utility function outside of the component to a utilities file if it's not using any component-specific logic. This can improve the testability and reusability of the function.